plugin-nvidia/src/nvml/features.rs

```diff
@@ -0,0 +1,288 @@
+        let mut available = Vec::new();
+        if self.total_energy_consumption {
+            available.push("total_energy_consumption");
+        }
+        if self.instant_power {
+            available.push("instant_power");
+        }
+        if self.major_utilization {
+            available.push("major_utilization");
+        }
+        if self.decoder_utilization {
+            available.push("decoder_utilization");
+        }
+        if self.encoder_utilization {
+            available.push("encoder_utilization");
+        }
+        if self.temperature_gpu {
+            available.push("temperature_gpu");
+        }
+        match self.running_compute_processes {
+            AvailableVersion::Latest => available.push("running_compute_processes(latest)"),
+            AvailableVersion::V2 => available.push("running_compute_processes(v2)"),
+            AvailableVersion::None => (),
+        };
+        match self.running_graphics_processes {
+            AvailableVersion::Latest => available.push("running_graphics_processes(latest)"),
+            AvailableVersion::V2 => available.push("running_graphics_processes(v2)"),
+            AvailableVersion::None => (),
+        };
+        write!(f, "{}", available.join(", "))
+    }
+}
+
+/// Verification of running compute processes validity on a GPU targeted `device` on operating system.
+///
+/// # Return
+/// 
+/// - `AvailableVersion` : Get and check data running graphics processes versioning (None, latest or 2.0)
+/// - An handled NVML library error about detection of running compute processes
+fn check_running_compute_processes(device: &Device) -> Result<AvailableVersion, NvmlError> {
+    match device.running_compute_processes() {
+        Ok(_) => Ok(AvailableVersion::Latest),
+        Err(NvmlError::FailedToLoadSymbol(_)) => match device.running_compute_processes_v2() {
+            Ok(_) => Ok(AvailableVersion::V2),
+            Err(e) => Err(e),
+        },
+        Err(e) => Err(e),
+    }
+}
+
+/// Verification of running graphics processes validity on a GPU targeted `device` on operating system.
+///
+/// # Return
+/// 
+/// - `AvailableVersion` : Get and check data running graphics processes versioning (None, latest or 2.0)
+/// - An handled NVML library error about detection of running graphics processes
+fn check_running_graphics_processes(device: &Device) -> Result<AvailableVersion, NvmlError> {
+    match device.running_graphics_processes() {
+        Ok(_) => Ok(AvailableVersion::Latest),
+        Err(NvmlError::FailedToLoadSymbol(_)) => match device.running_graphics_processes_v2() {
+            Ok(_) => Ok(AvailableVersion::V2),
+            Err(e) => Err(e),
+        },
+        Err(e) => Err(e),
+    }
+}
+
+/// Determine if a GPU parameter can be extracted or not to information system by NVML method.
+///
+/// # Return
+///
+/// - Return true if GPU NVML information exist.
+/// - Return false or NVML error message if GPU information does not exist or available.
```

Comment:
```suggestion
/// Checks if a feature is supported by the available GPU by inspecting the return type of an NVML function.
///
/// # Example
/// ```no_run
/// let device: &nvml_wrapper::Device = todo!();
/// let power_available = is_supported(device.power_usage())?;
/// ```
```